Sources/RInAppMessaging/Views/CarouselView.swift

```diff
@@ -122,6 +138,57 @@ extension CarouselView {
     func adjustHeight(height: CGFloat) -> CGFloat {
         return height < Constants.Carousel.minHeight ? Constants.Carousel.defaultHeight : height
     }
+
+    func startAutoScroll() {
+            stopAutoScroll()
+            guard !hasReachedLastImage else { return }
+            timer = Timer.scheduledTimer(timeInterval: 5.0, target: self, selector: #selector(scrollToNextItem), userInfo: nil, repeats: true)
+    }
+
+    func stopAutoScroll() {
+        timer?.invalidate()
+        timer = nil
+    }
+
+    @objc private func scrollToNextItem() {
+        guard !userIsScrolling, !carouselData.isEmpty else { return }
+
+        if currentIndex == carouselData.count - 1 {
+            stopAutoScroll() // Stop at the last image
+            hasReachedLastImage = true
+            return
+        }
+
+        let nextIndex = currentIndex + 1
+        let indexPath = IndexPath(item: nextIndex, section: 0)
+
+        collectionView.scrollToItem(at: indexPath, at: .centeredHorizontally, animated: true)
+        currentIndex = nextIndex
+    }
+
+    func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        userIsScrolling = true
```

Comment:
I don't think userIsScrolling property is required because when you invalidate and make timer as nil, auto scroll will be stopped

Reply:
yes, that is the purpose. When carousel loads it starts auto scrolling , if the user interacts and swipes back or forward it stops auto scrolling. It will start scrolling again after 5 seconds ,only if has not completely scrolled to the end once .

Reply:
`userIsScrolling`  Without this property also you can achieve the same result